iocore/eventsystem/test_Buffer.cc

```diff
@@ -91,23 +91,13 @@ reconfigure_diags()
 static void
 init_diags(const char *bdt, const char *bat)
 {
-  FILE *diags_log_fp;
   char diags_logpath[500];
   ink_strlcpy(diags_logpath, DIAGS_LOG_FILE, sizeof(diags_logpath));
 
-  diags_log_fp = fopen(diags_logpath, "w");
-  if (diags_log_fp) {
-    int status;
-    status = setvbuf(diags_log_fp, NULL, _IOLBF, 512);
-    if (status != 0) {
-      fclose(diags_log_fp);
-      diags_log_fp = NULL;
-    }
-  }
-
-  diags = new Diags(bdt, bat, diags_log_fp);
+  BaseLogFile *blf = new BaseLogFile(diags_logpath);
+  diags = new Diags(bdt, bat, blf);
 
-  if (diags_log_fp == NULL) {
+  if (blf && blf->m_fp == NULL) {
```

Comment:
`if (blf == NULL || blf->m_fp == NULL)` - you certainly don't want to continue on if `blf` is `NULL`.